File: app/src/androidTest/java/com/mycompany/myapp/TestUtils.java
package com.mycompany.myapp;

import android.app.Instrumentation;
import android.content.Context;
import android.support.test.InstrumentationRegistry;

import com.mycompany.myapp.app.MainApplication;

public class TestUtils {
    public static MainApplication getAppUnderTest() {
        Instrumentation instrumentation = InstrumentationRegistry.getInstrumentation();
        Context targetContext = instrumentation.getTargetContext();
        return (MainApplication) targetContext.getApplicationContext();
    }
}

File: app/src/androidTest/java/com/mycompany/myapp/TestUtils.kt
package com.mycompany.myapp

import androidx.test.platform.app.InstrumentationRegistry
import com.mycompany.myapp.app.MainApplication

object TestUtils {
    fun getAppUnderTest(): MainApplication {
        val instrumentation = InstrumentationRegistry.getInstrumentation()
        val targetContext = instrumentation.targetContext
        return targetContext.applicationContext as MainApplication
    }
}
